dfastmi/batch/ConfigurationCheckerLegacy.py

```diff
@@ -0,0 +1,426 @@
+        from configuration in batch mode (no user interaction).
+
+        Arguments
+        ---------
+        config : configparser.ConfigParser
+            Configuration of the analysis to be run.
+        q_threshold : Optional[float]
+            River discharge at which the measure becomes active 
+        q_levels : 
+            Characteristic discharges used by algorithm [m3/s].
+        Results
+        -------
+        q_bankfull : float
+            River discharge at which the measure is bankfull [m3/s].
+        """
+        if q_threshold is None or q_threshold < q_levels[1]:
+            q_bankfull = config.get("General", "Qbankfull", fallback="")
+            if self._is_float_str(q_bankfull):
+                q_bankfull = float(q_bankfull)
+            else:
+                q_bankfull = 0
+        else:
+            q_bankfull = 0
+        return q_bankfull
+
+    def _get_q_threshold_from_config(self, config):
+        """
+        Get the simulation discharge threshold from configuration in batch mode (no user interaction).
+
+        Arguments
+        ---------
+        config : configparser.ConfigParser
+            Configuration of the analysis to be run.
+        
+        Results
+        -------
+        q_threshold : Optional[float]
+            River discharge at which the measure becomes active [m3/s].
+        """
+
+        q_threshold = config.get("General", "Qthreshold", fallback="")
+        if self._is_float_str(q_threshold):
+            q_threshold = float(q_threshold)
+        else:
+            q_threshold = None
+        return q_threshold
+
+
+    def _is_float_str(self, string:str) -> bool:
+        """
+        Check if a string represents a (floating point) number.
+
+        Arguments
+        ---------
+        string : str
+            The string to be checked.
+
+        Returns
+        -------
+        success : bool
+            Boolean indicating whether the the string can be converted to a number or not.
+        """
+        try:
+            float(string)
+            return True
+        except ValueError:
+            return False
+
+    def _check_configuration_cond_waqua(
+            self,
+            config: configparser.ConfigParser,
+            i : int) -> bool:
+        """
+        Check validity of one condition of a version 1 analysis configuration using WAQUA results.
+
+        Arguments
+        ---------
+        config : configparser.ConfigParser
+            Configuration for the D-FAST Morphological Impact analysis.
+        i : int
+            Flow condition to be checked.            
+
+        Returns
+        -------
+        success : bool
+            Boolean indicating whether the D-FAST MI analysis configuration is valid.
+        """
+        cond = "Q" + str(i+1)
+        # condition block to be checked.
+        # condition block must be specified since it must contain the Reference and WithMeasure file names
+        if cond not in config.sections():
```

Comment:
check if this is double with the things below.

Reply:
refactored